### DISCUSSION TITLE
Supabase rc 2 typescript types from remote db

### DISCUSSION THREAD

**Discussion by vavdav** [supabase/supabase#9379] [Questions] [answered]
For supabase 1 there was open-api to generate types for typescript which is pretty clear: https://supabase.com/docs/guides/api/generating-types 

`npx openapi-typescript https://your-project.supabase.co/rest/v1/?apikey=your-anon-key --output types/supabase.ts`

But I am not able to use them with supabase rc 2.
<img width="823" alt="Screen Shot 2022-10-06 at 23 38 53" src="https://user-images.githubusercontent.com/215969/194424407-e41596e7-5388-4552-9ee9-85e8cca1cc7a.png">

So I assume I need to follow the new documentation, but there is no description of how to do it. There is only one for local DB. Plus I found `--db-url` flag but not more info regarding that.

https://supabase.com/docs/reference/javascript/next/typescript-support

I have tried this command: `npx supabase gen types typescript --db-url https://your-project.supabase.co > src/database.types.ts`, but I am getting error: `Error: URL is not a valid Supabase connection string.`

Thank you for help

**Comment by ThatXliner** [comment]
The connection URL should start with `postgres`

**Comment by vavdav** [comment]
@ThatXliner this helped to get to another error.

With command:
```npx supabase gen types typescript --db-url postgresql://postgres:[PASSWORD]@db.*****.supabase.co:5432/postgres > src/database.types.ts§```

I get error: `Error: error running container`

**Reply by ThatXliner** [reply]
1. [It appears to me that only the open-api command works properly](https://github.com/supabase/supabase/discussions/9419#discussioncomment-3831241)
2. I remember getting that error but I'm not sure how I solved it. I think I nuked the folder and did `supabase init` and `supabase start` again.

**Comment by hyunbinseo** [comment, marked as answer]
This worked for me.

```
npm i supabase -D
```
```
npx supabase login
You can generate an access token from https://app.supabase.com/account/tokens
Enter your access token: token-goes-here
Finished supabase login.
```
```
npx supabase gen types typescript --project-id id-goes-here > lib/database.types.ts
```

**Reply by soedirgo** [reply]
Marking this as answer - sorry for the wrinkles. The [guide](https://supabase.com/docs/guides/api/generating-types) has been updated to reflect this.

**Reply by fergusmeiklejohn** [reply]
this doesn't work anymore because the `--project-id` flag has been removed from the CLI (but not from the docs)

now there is only the `--db-url` flag, which fails with `Error: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?` (why would Docker need to be running to connect to a remote DB?) or if docker is running `Error: error running container`

**Reply by hyunbinseo** [reply]
@fergusmeiklejohn The `--project-id` flag still works in [v1.11.3 (latest)](https://github.com/supabase/cli/releases/tag/v1.11.3) and [v1.11.2](https://github.com/supabase/cli/releases/tag/v1.11.2).

**Reply by o-az** [reply]
and [v1.75.3 (latest)](https://github.com/supabase/cli/releases/tag/v1.75.3)
